### ISSUE TITLE
Which result is more reliable?

### ISSUE THREAD

**Issue by wjw-yj** [shahab-sarmashghi/RESPECT#21]
Dear Shahab,
First of all, many thanks for making this tool available.
We extracted 1x and 2x data and used KMC software to generate different *.hist files with kmer values of 21, 24, 27, 30, 33, 36, 39, 42, 45, 48, and 52 (11 different kmer values) to estimate the genome size. However, we found two problems:
Ⅰ. The genome size estimated from the hist files generated by the 2x data is larger than that estimated from the 1x data. Should we choose the result of 2x or 1x?
Ⅱ. In the 1x data, the genome sizes estimated with different kmers are not the same, and the genome size decreases slightly as the kmer value increases (kmer = 21, GS = 12019916620; kmer = 45, GS = 9404517373). Nevertheless, we can see that for the same 1x data, different genome sizes are produced with different kmers. At what kmer value is the genome size closest to the real value, or within what range? Could you please answer my questions? Thank you very much.

Sincerely，
wjw

**Comment by shahab-sarmashghi** [comment]
Hi,

We didn't tested it with k-mer sizes larger than 31. Theoretically I don't see a problem with larger k-mer sizes but they might need tweaking some of the algorithm parameters from their default values. I would suggest k=31 but this is just because we didn't test the larger values. 

As for coverage, I would use 2X. In our [simulations](https://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=10.1371/journal.pcbi.1009449.s019), the accuracy slightly improved as we increased the coverage from 0.5X to 4X.

**Comment by wjw-yj** [comment]
Dear Shahab,
Thank you very much for your reply. I understand that the data error of the 4x data is smaller. My question is, which k - mer value can estimate the genome size more accurately? Is k - mer = 31 the closest to the true value? 


Sincerely，
wjw

**Comment by shahab-sarmashghi** [comment]
Yes, I suggest using k=31. My point was theoretically you should get similar results with larger k-mer sizes but since there are many moving parts in the algorithm and we have optimized it for k=31, that might be the best choice.
